packages/expo-permissions/android/src/main/java/expo/modules/permissions/PermissionsModule.kt

```diff
@@ -2,253 +2,40 @@
 
 package expo.modules.permissions
 
-import android.Manifest
-import android.annotation.TargetApi
 import android.content.Context
-import android.content.Intent
-import android.content.pm.PackageManager
-import android.net.Uri
-import android.os.Build
-import android.os.Bundle
-import android.provider.Settings
-import expo.modules.permissions.requesters.*
-
 import org.unimodules.core.ExportedModule
 import org.unimodules.core.ModuleRegistry
 import org.unimodules.core.Promise
-import org.unimodules.core.interfaces.ActivityProvider
 import org.unimodules.core.interfaces.ExpoMethod
-import org.unimodules.core.interfaces.LifecycleEventListener
-import org.unimodules.core.interfaces.services.UIManager
 import org.unimodules.interfaces.permissions.Permissions
 
-import kotlin.collections.ArrayList
-import kotlin.collections.HashSet
-import kotlin.properties.Delegates
-
-internal const val EXPIRES_KEY = "expires"
-internal const val STATUS_KEY = "status"
-internal const val GRANTED_VALUE = "granted"
-internal const val DENIED_VALUE = "denied"
-internal const val UNDETERMINED_VALUE = "undetermined"
-internal const val ERROR_TAG = "E_PERMISSIONS"
-
-internal const val PERMISSION_EXPIRES_NEVER = "never"
-
-class PermissionsModule(context: Context) : ExportedModule(context), LifecycleEventListener {
-
-  private var mActivityProvider: ActivityProvider? = null
-
-  // state holders for asking for writing permissions
-  private var mWritingPermissionBeingAsked = false // change this directly before calling corresponding startActivity
-  private var mAskAsyncPromise: Promise? = null
-  private var mAskAsyncRequestedPermissionsTypes: ArrayList<String>? = null
-  private var mAskAsyncPermissionsTypesToBeAsked: ArrayList<String>? = null
-
-  private var mRequesters: Map<String, PermissionRequester> by Delegates.notNull()
 
-  companion object {
-    private val mPermissionsAskedFor = HashSet<String>()
-    fun didAsk(permission: String): Boolean {
-      return mPermissionsAskedFor.contains(permission)
-    }
-
-    private var mPermissions: Permissions? = null
-    fun getPermissionService(): Permissions {
-      return mPermissions ?: throw IllegalStateException("No Permissions module present.")
-    }
-  }
+class PermissionsModule(context: Context) : ExportedModule(context) {
+  private var mPermissions: Permissions? = null
```

Comment:
I'd like to suggest more convenient construction https://kotlinlang.org/docs/reference/properties.html#late-initialized-properties-and-variables